rules.mk

```diff
@@ -106,7 +106,9 @@ KERNCFLAGS+=    -mno-sse \
 		-mno-sse2
 endif
 KERNCFLAGS+=	-fno-omit-frame-pointer
-KERNLDFLAGS=	--gc-sections -n
+KERNLDFLAGS=	--gc-sections -n \
+		-u ipaddr_ntoa_r -u dns_gethostbyname -u netif_input -u netif_remove \
```

Comment:
This could be way too much hassle so feel free to ignore, but do you think there's a way to autogenerate these? That might even remove the necessity of the special macro and text section. For example, 'nm -u' on the klib could be used to generate a list of names which could be dumped to a file as an EXTERN linker script command (which is supposed to be equivalent of -u) that could be included by the kernel linker script. Provided klibs are built before the kernel, of course.

Reply:
Now the list of dynamic relocation entries found in each klib is retrieved via objdump and transformed (using sed scripts) into a set of EXTERN linker commands, which are put in an auto-generated file (output/klib/klib-syms.lds) that is included by the kernel linker scripts. so we no longer need to manually add `-u` linker options, nor to use the KLIB_EXPORT #define and the .text.klib linker section.

Reply:
> Now the list of dynamic relocation entries found in each klib is retrieved via objdump and transformed (using sed scripts) into a set of EXTERN linker commands, which are put in an auto-generated file (output/klib/klib-syms.lds) that is included by the kernel linker scripts. so we no longer need to manually add `-u` linker options, nor to use the KLIB_EXPORT #define and the .text.klib linker section.

This is nice. As long as klibs are always built at the same time as the kernel (and within the same tree), I see no problem with this. And if there might ever be an exception to that, we could include a file with a list of symbols to always EXTERN.

Reply:
Great, thanks!